lib/tasks/match_identity_with_professional_organization.rake

```diff
@@ -0,0 +1,208 @@
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS~
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR~
+# TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.~
+
+desc "Match Identity with Professional Organization"
+task :match_identity_with_professional_organization => :environment do
+
+  #### DEPARTMENT ####
+  puts "START"
+  puts "Already matched Identities:"
+  Identity.all.where.not(professional_organization_id: nil).map(&:id)
+  puts "Number of Unmatched Identities:"
+  puts Identity.all.where(professional_organization_id: nil).count
+
+  one_department_match = Hash.new
+  more_than_one_department_match = Hash.new
+
+  identity_departments = Identity.all.map(&:department).uniq
+
+  # Remove nil and empty
+  identity_departments.delete(nil)
+  identity_departments.delete("")
+
+  # Task: to match Identity to Professional Organization.
+  # There was not a concrete map on how to do this.
+  # Match was based on partial word equality.
+  # First matched the departments (the lowest level), 
+  # then matched at the college level,
+  # finally matched at the institution level
+  @departments_that_need_review = []
+  identity_departments.each do |department|
+    prof_org = ProfessionalOrganization.where("LOWER(professional_organizations.name) LIKE LOWER('%#{department}%')").where(org_type: 'department')
+    if !prof_org.empty?
+      if prof_org.count == 1
+        one_department_match[department] = prof_org.first.id
+      else
+        more_than_one_department_match[department] = prof_org.map(&:id)
+      end
+    else
+      @departments_that_need_review << department
+    end
+  end
+
+  # @departments_that_need_review are departments with no partial word matches.
+  # These were reviewed with Wenjun and she supplied the match for 
+  # "pharmaceutical_and_biomedical_sciences", "orthopaedic_surgery", "pharmacy_and_clinical_sciences", "cell_biology_and_anatomy", "craniofacial_biology"
+
+  @departments_that_need_review.delete("pharmaceutical_and_biomedical_sciences") ## Addressed, further down
+  @departments_that_need_review.delete("orthopaedic_surgery")
+  @departments_that_need_review.delete("pharmacy_and_clinical_sciences")
+  @departments_that_need_review.delete("cell_biology_and_anatomy")
+  @departments_that_need_review.delete("craniofacial_biology")
+
+  # These are the remaining departments that do not have matches.
+  # Wenjun decided that in these cases, we match the college, and if the
+  # college doesn't match either, we match to the institution 
+  puts "DEPARTMENTS THAT NEED REVIEW: "
+  puts @departments_that_need_review.inspect
+
+  one_department_match["orthopaedic_surgery"] = 52
```

Comment:
You could do an iteration over a hash here if you wanted
{"pharmacy_and_clinical-sciences" => 122, "cell_biolgy_and_anatomy" => 60, "craniofacial_biology" => 11, "medicine" => 45, "radiology" => 59, "surgery" => 61, "urology" => 62}.each do |k, v|
.  one_department_match[k] = v
end